src/core/dbus-unit.c

```diff
@@ -1527,15 +1531,11 @@ int bus_unit_method_attach_processes(sd_bus_message *message, void *userdata, sd
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_ACCESS_DENIED, "Process " PID_FMT " not owned by target unit's UID. Refusing.", pid);
                 }
 
-                if (!pids) {
-                        pids = set_new(NULL);
-                        if (!pids)
-                                return -ENOMEM;
-                }
-
-                r = set_put(pids, PID_TO_PTR(pid));
+                r = set_ensure_put(&pids, &pidref_hash_ops, pidref);
                 if (r < 0)
                         return r;
+
+                TAKE_PTR(pidref);
```

Comment:
`_consume` variant? Or am I missing something that makes it unavailable here?

Reply:
indeed. fixed.